### PR TITLE
Accelerate anchors generator in early steps

### PR DESCRIPTION
**Pull request by yd8534976**
## Motivation and Context

In early training steps of YOLOv2, all ground truth anchors will be generated for computing global coordinate loss. Current implementation uses 3 loops, it is not efficient especially when number of cells increases.
## Description

I reimplemented this part without loops. It may be not obvious under 5x5 cells, so I also tested it under 20x20 cells.
## How has this been tested?
I tested `lm_fyolo` with config file `lm_fyolo_quantize_pascalvoc_2007_2012.py`, it is slightly faster than before (maybe under the inference of server environments, so I also test it using jupyter notebook as below), and mAP is almost same as before.

## Screenshots (if appropriate):
Screen Shot 2019-04-25 at 14 38 17

Screen Shot 2019-04-25 at 14 39 17
Screen Shot 2019-04-25 at 14 39 28

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by iizukak** [comment]
run test

**Comment by iizukak** [comment]
Thanks @yd8534976 , I'll test soon.

**Comment by iizukak** [comment]
ok to test

**Comment by ruimashita** [comment]
@yd8534976 Thanks 👍👍👍

**Comment by yd8534976** [comment]
@ruimashita Thanks for review and advice! I'll check if `YoloV2.offset_boxes()` is reusable.

**Comment by yd8534976** [comment]
- For better efficiency, I take anchor generator out of batch loop.
- To reuse 'YoloV2._offset_boxes()`, I have to change it to public and static method. I'm not sure if this change will affect inference time.
- I think we should implement a `AnchorGenerator` class in `lmnet` which can be reused by future models.

A simple test using `google colaboratory`:
https://colab.research.google.com/drive/1WSCo4jn3R6426cpcrx1twlvwYlVVJExF#scrollTo=xTYeo_iE_OOX&uniqifier=2

**Comment by yd8534976** [comment]
@iizukak , thanks for reviewing!

File: lmnet/lmnet/networks/object_detection/yolo_v2.py [hunk 1]
```diff
@@ -1253,27 +1262,30 @@ def __calculate_truth_and_maskes(
...
+            stride_x = image_size[1] / num_cell_x
+            stride_y = image_size[0] / num_cell_y
+
+            cell_gt_boxes[:, :, :, :, 0] = (offset_x + 0.5) * stride_x
+            cell_gt_boxes[:, :, :, :, 1] = (offset_y + 0.5) * stride_y
+            cell_gt_boxes[:, :, :, :, 2] = stride_x * offset_w
+            cell_gt_boxes[:, :, :, :, 3] = stride_y * offset_h
+            cell_gt_boxes[:, :, :, :, 4] = -1
+
+            coordinate_maskes += 1  # True
```
Comment (ruimashita) [inline lmnet/lmnet/networks/object_detection/yolo_v2.py]:
```suggestion
            coordinate_maskes[:] = 1  # True
```

`+=` force to confirm `coordinate_maskes` before values.

**Comment by ruimashita** [comment]
@yd8534976 
LGTM 😄 Thanks 👍👍👍
